src/java_bytecode/java_bytecode_parser.cpp

```diff
@@ -1381,6 +1403,101 @@ void java_bytecode_parsert::rclass_attribute(classt &parsed_class)
+      }
+      else
+      {
+        // try parsing bootstrap method handle
+        if(num_bootstrap_arguments == 3)
+        {
+
+          // each entry contains a MethodHandle structure
+          // u2 tag
+          // u2 reference kind which must be in the range from 1 to 9
+          // u2 reference index into the constant pool
+          //
+          // reference kinds use the following
+          // 1 to 4 must point to a CONSTANT_Fieldref structure
+          // 5 or 8 must point to a CONSTANT_Methodref structure
+          // 6 or 7 must point to a CONSTANT_Methodref or
+          // CONSTANT_InterfaceMethodref structure, if the class file version number
+          // is 52.0 or above, to a CONSTANT_Methodref only in the case of less than
+          // 52.0
+          // 9 must point to a CONSTANT_InterfaceMethodref structure
+
+          // the index must point to a CONSTANT_String
+          //                           CONSTANT_Class
+          //                           CONSTANT_Integer
+          //                           CONSTANT_Long
+          //                           CONSTANT_Float
+          //                           CONSTANT_Double
+          //                           CONSTANT_MethodHandle
+          //                           CONSTANT_MethodType
+          u2 arg_index1 = read_u2();
+          u2 arg_index2 = read_u2();
+          u2 arg_index3 = read_u2();
+
+          const pool_entryt &arg1 = pool_entry(arg_index1);
+          const pool_entryt &arg2 = pool_entry(arg_index2);
+          const pool_entryt &arg3 = pool_entry(arg_index3);
+
+          // std::cout << "bootstrap arg1 tag " << std::to_string(arg1.tag) << std::endl;
+          // std::cout << "bootstrap arg2 tag " << std::to_string(arg2.tag) << std::endl;
+          // std::cout << "bootstrap arg3 tag " << std::to_string(arg3.tag) << std::endl;
+
+          if(!(arg1.tag == CONSTANT_MethodType &&
+               arg2.tag == CONSTANT_MethodHandle &&
+               arg3.tag == CONSTANT_MethodType))
+            return;
+
+          // std::cout << "MethodType arg1 " << id2string(pool_entry(arg1.ref1).s)
+          //           << "\nMethodType arg2 " << id2string(pool_entry(arg3.ref1).s)
+          //           << std::endl;
+          // std::cout << std::endl;
+
+          lambda_method_handlet real_handle;
+          parse_methodhandle(arg_index2, 0, real_handle);
+          if(real_handle.handle_type != method_handle_typet::LAMBDA_METHOD_HANDLE)
+          {
+            lambda_method_handlet empty_handle;
+            lambda_method_handles.push_back(empty_handle);
+            error() << "ERROR: could not parse lambda function method handle" << eom;
+          }
+          else
+          {
+            lambda_method_handles.push_back(real_handle);
+          }
+        }
+        else
+        {
+          // skip arguments here
+          for(size_t i = 0; i < num_bootstrap_arguments; i++)
+            read_u2();
```

Comment:
Actually it may be better to use a call to skip_bytes (or a new `skip_u2s`) method rather than writing an additional loop each time you want to skip things.